x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -74,6 +74,33 @@ async function syncDataForAllLayers(getState, dispatch, dataFilters) {
   await Promise.all(syncs);
 }
 
+export function trackCurrentLayerState(layerId) {
+  return {
```

Comment:
This should also dispatch removeTrackedLayerStateForSelectedLayer to ensure the previous selected layer state tracker is cleaned up. This is important in the instance where the layer panel is open and instead of clicking `cancel`, the user clicks another layer name in the TOC. That will update selected layer by not call `removeTrackedLayerStateForSelectedLayer `

Reply:
I think this is already handled. Could you double check (?)

Reply:
It is but in the wrong place. Dispatching an action should not have side-effects. see "move to action creator middleware. Actions should not have side effects" comment below

Reply:
RE: actions & side effects. Interesting @nreese, in this case I agree, but I don't always think it's unreasonable for different actions to have overlapping, or occasionally the exact same, effects on the store. There are any number of competing conventions here though, maybe we should discuss this one internally and come to some agreement.